### PR TITLE
Patch #3

### PR DESCRIPTION
**Pull request by armandojaga**
adding precompiled headers
adding themes
adding event system
renaming the LOG function
fixing bug when exiting from menu
renaming folders
fixing typo on gameobject
adding JetBrains mono as default font

### CODE REVIEW

File: Source/src/Modules/ModuleEditor.h [hunk 1]
```diff
@@ -99,5 +109,7 @@ namespace Hachiko
         WindowResource w_resource;
```
Comment (baransrc) [inline Source/src/Modules/ModuleEditor.h:109]: Again in the next iteration, we can make these names less cryptic

File: Source/src/Modules/ModuleInput.h [hunk 2]
```diff
@@ -63,7 +63,7 @@ namespace Hachiko
 
         KeyState* keyboard = nullptr;
         Uint32 mouse{};
-        SDL_Keymod keymods;
+        SDL_Keymod keymods{};
```
Comment (baransrc) [inline Source/src/Modules/ModuleInput.h:66]: We should have all the initialization of the member fields in the default constructor in the order of their declarations. In the next iteration maybe we can have that consitency update in all classes.

File: Source/src/UI/Window.cpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#include "Window.h"
+#include "core/hepch.h"
 
-Hachiko::Window::Window(const char* name, bool active) :
+Hachiko::Window::Window(const char* name, const bool active) :
```
Comment (baransrc) [inline Source/src/UI/Window.cpp:3]: What's the point of having const in bool? Is it to further indicate to the maintainer of this method to just use the flag as readonly?

**Review by baransrc — APPROVED** [review]
I left some notes for possible things we can do in the next patch, as always, amazing work, thank you so much for your effort!!

**Review by alvarosoppelsa — APPROVED** [review]

File: Source/src/Core/hepch.cpp [hunk 4]
```diff
@@ -0,0 +1 @@
+#include "core/hepch.h"
```
[the remark this answers is not in the document]
Comment (armandojaga) [reply Source/src/Core/hepch.cpp:1]: yes, this is the precompiled header, is how visual studio works

File: Source/src/Importers/MaterialImporter.cpp [hunk 5]
```diff
@@ -46,7 +46,7 @@ Hachiko::ResourceMaterial* Hachiko::MaterialImporter::Material::Import(const aiM
 
     for (std::string path : search_paths)
     {
-        r_material->diffuse = App->texture->Load(path.c_str());
+        r_material->diffuse = ModuleTexture::Load(path.c_str());
```
[the remark this answers is not in the document]
Comment (armandojaga) [reply Source/src/Importers/MaterialImporter.cpp:49]: there's a lot of refactoring left to do, yes

File: Source/src/UI/Window.cpp (hunk 3 above)
Comment (armandojaga) [reply Source/src/UI/Window.cpp:3]: this is a futile for this data type, but it is a format that was applied project-wise

**Pull request merged**